src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1379,27 +1379,6 @@ void JNIGlobalsDumper::do_oop(oop* obj_p) {
   }
 };
 
-
-// Support class used to generate HPROF_GC_ROOT_MONITOR_USED records
-
-class MonitorUsedDumper : public OopClosure {
- private:
-  DumpWriter* _writer;
-  DumpWriter* writer() const                { return _writer; }
- public:
-  MonitorUsedDumper(DumpWriter* writer) {
-    _writer = writer;
-  }
-  void do_oop(oop* obj_p) {
-    u4 size = 1 + sizeof(address);
-    writer()->start_sub_record(HPROF_GC_ROOT_MONITOR_USED, size);
```

Comment:
Same answer as above. No longer a strong root, so can't report it as such.

Reply:
We no longer have a root in the ObjectMonitor so no we don't have
to dump these is my take. I believe @fisk concurs with that reasoning.

Reply:
Agreed.

Reply:
Ok, that's good then!

Reply:
Thanks for confirmation.

Reply:
I've taken a first pass at creating a CSR:
JDK-8253121 migrate ObjectMonitor::_object to OopStorage
https://bugs.openjdk.java.net/browse/JDK-8253121

Reply:
I've looked at the CSR and added myself as a reviewer.
We had a slack chat with Dan, and I agree that with a weak handle it would be racy/unsafe for JVMTI_HEAP_REFERENCE_MONITOR calls back to be called. The ObjectMonitors do not pin objects anymore (there are no strong refs from them), so it has to be okay to skip reporting the JVMTI_HEAP_REFERENCE_MONITOR and and JVMTI_HEAP_ROOT_MONITOR (old Heap Walking API) reference types. The JVMTI does not need an update as other VM implementations can still report these reference types. Alan added a comment to the CSR saying that memory profiling tools that use the JVMTI functions (FollowReferences with jvmtiHeapReferenceCallback or IterateOverReachableObjects with jvmtiHeapRootCallback) to iterate over the heap should not have a compatibility impact as these reference types are just informational but adding a release note can be useful.

Reply:
Slight clarification. Serguei and I were discussing whether we could continue
to make JVMTI_HEAP_REFERENCE_MONITOR call backs or emit
HPROF_GC_ROOT_MONITOR_USED entries in heap dump output as a
way to ease the transition phase of getting used to these things going
away. My answer was that we could do that but it would racy and unsafe
due to the ObjectMonitor's object being GC'ed.

It's also incorrect to make JVMTI_HEAP_REFERENCE_MONITOR call backs
or emit HPROF_GC_ROOT_MONITOR_USED entries in heap dump once
the ObjectMonitor's object ref becomes a weak handle. That weak handle
no longer prevents the associated object from being GC'ed so it is no
longer a strong root.

See Erik's comment above: https://github.com/openjdk/jdk/pull/135#discussion_r487291746